.circleci/config.yml

```diff
@@ -0,0 +1,96 @@
+version: 2.1
+jobs:
+  build:
+    docker:
+      - image: "golang:1.12.0-alpine3.9"
+    steps:
+      - run: apk add --update --no-cache ca-certificates git
+      - checkout
+      - run: go mod download
+      - run: go build .
+  tileserver-base-env-vars:
+  #FIXME - find a way to inject env vars directly from workflow defn 
+    docker:
+      - image: "alpine:3.9"
+    steps:
+      - run: apk add --no-cache ca-certificates
+      - run: mkdir -p /tmp/workspace
+      - run: echo "export DOCKER_FILE=Dockerfile" >> /tmp/workspace/env-vars
+      - run: echo "export SERVICE_NAME=tileserver" >> /tmp/workspace/env-vars
+      - persist_to_workspace:
+          root: /tmp/workspace
+          paths:
+            - env-vars
+  tileserver-tag-env-vars:
+    docker:
+      - image: "alpine:3.9"
+    steps:
+      - run: apk add --no-cache ca-certificates
+      - attach_workspace:
+          at: /tmp/workspace
+      - run: echo "export TAG=$CIRCLE_TAG" >> /tmp/workspace/env-vars
+      - persist_to_workspace:
+          root: /tmp/workspace
+          paths:
+            - env-vars
+  tileserver-branch-env-vars:
+    docker:
+      - image: "alpine:3.9"
+    steps:
+      - run: apk add --no-cache ca-certificates
+      - attach_workspace:
+          at: /tmp/workspace
+      - run: echo "export TAG=$CIRCLE_SHA1" >> /tmp/workspace/env-vars
+      - persist_to_workspace:
+          root: /tmp/workspace
+          paths:
+            - env-vars
+  upload:
+    docker: 
+      #FIXME: find image that has docker and also make sure it has circleci creds
+      - image: "circleci/golang:1.8.6-jessie"
```

Comment:
You should probably be running this on a more recent version of Debian (stretch or buster).